src/main/kotlin/uk/gov/dluhc/printapi/service/temporarycertificate/CertificatePdfTemplateDetailsFactory.kt

```diff
@@ -0,0 +1,99 @@
+    @Value("\${temporary-certificate.certificate-pdf.english.images.voter-photo.absolute-x}") private val voterPhotoAbsoluteXEnPlaceholder: Float,
+    @Value("\${temporary-certificate.certificate-pdf.english.images.voter-photo.absolute-y}") private val voterPhotoAbsoluteYEnPlaceholder: Float,
+    @Value("\${temporary-certificate.certificate-pdf.english.images.voter-photo.fit-width}") private val voterPhotoFitWidthEnPlaceholder: Float,
+    @Value("\${temporary-certificate.certificate-pdf.english.images.voter-photo.fit-height}") private val voterPhotoFitHeightEnPlaceholder: Float,
+    @Value("\${temporary-certificate.certificate-pdf.welsh.path}") private val pdfTemplateWelsh: String,
+    @Value("\${temporary-certificate.certificate-pdf.welsh.placeholder.elector-name}") private val electorNameCyPlaceholder: String,
+    @Value("\${temporary-certificate.certificate-pdf.welsh.placeholder.local-authority-name}") private val localAuthorityNameCyPlaceholder: String,
+    @Value("\${temporary-certificate.certificate-pdf.welsh.placeholder.date-of-issue}") private val dateOfIssueCyPlaceholder: String,
+    @Value("\${temporary-certificate.certificate-pdf.welsh.placeholder.valid-on-date}") private val validOnDateCyPlaceholder: String,
+    @Value("\${temporary-certificate.certificate-pdf.welsh.placeholder.certificate-number}") private val certificateNumberCyPlaceholder: String,
+    @Value("\${temporary-certificate.certificate-pdf.welsh.images.voter-photo.page-number}") private val voterPhotoPageNoCyPlaceholder: Int,
+    @Value("\${temporary-certificate.certificate-pdf.welsh.images.voter-photo.absolute-x}") private val voterPhotoAbsoluteXCyPlaceholder: Float,
+    @Value("\${temporary-certificate.certificate-pdf.welsh.images.voter-photo.absolute-y}") private val voterPhotoAbsoluteYCyPlaceholder: Float,
+    @Value("\${temporary-certificate.certificate-pdf.welsh.images.voter-photo.fit-width}") private val voterPhotoFitWidthCyPlaceholder: Float,
+    @Value("\${temporary-certificate.certificate-pdf.welsh.images.voter-photo.fit-height}") private val voterPhotoFitHeightCyPlaceholder: Float,
+) {
+    companion object {
+        private val DATE_TIME_FORMATTER = DateTimeFormatter.ofPattern("dd/MM/yyyy")
+    }
+
+    fun getTemplateDetails(certificate: TemporaryCertificate): TemplateDetails {
+        val voterPhotoBytes = getPhotoBytes(certificate.photoLocationArn!!)
+        return if (isWalesCode(certificate.gssCode!!)) {
+            val images = listOf(getVoterPhotoImageCy(voterPhotoBytes))
+            TemplateDetails(pdfTemplateWelsh, getWelshTemplatePlaceholders(certificate), images)
+        } else {
+            val images = listOf(getVoterPhotoImageEn(voterPhotoBytes))
+            TemplateDetails(pdfTemplateEnglish, getEnglishTemplatePlaceholders(certificate), images)
+        }
+    }
+
+    private fun getPhotoBytes(photoLocationArn: String): ByteArray {
+        // TODO get photo from S3
```

Comment:
To keep this PR a bit smaller, will include integrating with S3 when adding the controller endpoint and integration tests